kostyor/rest_api.py

```diff
@@ -49,11 +49,10 @@ def update_cluster(cluster_id):
 
 @app.route('/upgrade-status/<cluster_id>')
 def get_upgrade_status(cluster_id):
-    upgrade = db_api.get_upgrade_by_cluster(cluster_id)
-    if upgrade:
-        full_url = url_for('.get_upgrade', upgrade_id=upgrade['id'])
-        return redirect(full_url)
-    else:
+    try:
+        upgrade = db_api.get_upgrade_by_cluster(cluster_id)
+        return redirect(url_for('.get_upgrade', upgrade_id=upgrade['id']))
+    except:
```

Comment:
It's generally not a good idea to use pokemon exception handling. There could be plenty of reasons why called function throws an error and responding with *Upgrade for cluster not found* in this case may be wrong.

Could you please introduce domain-specific exceptions in some `exceptions.py` module and use it for throwing and catching instead?